pages/index.js

```diff
@@ -67,26 +104,77 @@ export default function Home({ posts, status }) {
-          <input
-            className="!border-0 !bg-gray-100"
-            placeholder="Search Blogs"
-            type="text"
-            value={keyword}
-            onChange={(e) => {
-              setKeyword(e.target.value);
-              setResults(searchBlogs(posts, e.target.value));
-            }}
-          />
+
+        <div className="flex flex-col lg:flex-row space-y-8 lg:space-y-0 lg:justify-between lg:items-center mb-16">
+          <div className="lg:basis-8/12 h-10 border-b border-[#e6e6e6]">
+            <Swiper
+              navigation={true}
+              onClick={(swiper) => setActiveIndex(swiper.clickedIndex)}
+              modules={[Navigation]}
+              breakpoints={{
+                0: {
+                  slidesPerView: 2,
+                },
+                576: {
+                  slidesPerView: 3,
+                },
+                1200: {
+                  slidesPerView: 5,
+                },
+              }}
+            >
+              <SwiperSlide
+                data-history="slide 0"
+                onClick={filterBlogs}
+                className={`pb-[0.9rem] capitalize ${
+                  activeIndex == "0"
+                    ? "after:absolute after:bottom-0 after:left-0 after:w-full after:h-0.5 after:bg-gradient-to-r from-[#f2709c] to-[#ff9472] canopas-gradient-text "
+                    : ""
+                }`}
+              >
+                {category}
+              </SwiperSlide>
+
+              {categories.map((category, index) => {
+                index = index + 1;
+                return (
+                  <SwiperSlide
+                    key={category.id}
+                    data-history={"slide " + category.id}
+                    onClick={filterBlogs}
+                    className={`pb-[0.9rem] capitalize ${
+                      activeIndex == index
+                        ? "after:absolute after:bottom-0 after:left-0 after:w-full after:h-0.5 after:bg-gradient-to-r from-[#f2709c] to-[#ff9472] canopas-gradient-text "
+                        : ""
+                    }`}
+                  >
+                    {category.attributes.name}
+                  </SwiperSlide>
+                );
+              })}
+            </Swiper>
+          </div>
+
+          <div className="flex flex-row items-center lg:basis-3/12 w-80 rounded-[10px] !bg-gray-100 pl-3">
+            <span>
+              <i className="w-16 h-16 rounded-full text-gray-500 cursor-pointer">
+                <FontAwesomeIcon
+                  icon={faMagnifyingGlass}
+                  className="pr-1 text-sm"
+                />
+              </i>
+            </span>
+            <input
+              className="!border-0 !bg-gray-100"
+              placeholder="Search Blogs"
+              type="text"
+              value={keyword}
+              onChange={(e) => {
+                setKeyword(e.target.value);
+                setResults(searchBlogs(categoryPosts, e.target.value));
```

Comment:
I think no need to pass `categoryPosts` in params, as we can use it directly in method